source/val/validation_state.h

```diff
@@ -155,32 +155,26 @@ class ValidationState_t {
   /// instruction
   bool in_block() const;
 
-  /// Registers the given <id> as an Entry Point with |execution_model|.
-  void RegisterEntryPointId(const uint32_t id,
-                            SpvExecutionModel execution_model) {
+  /// Registers |id| as an entry point with |execution_model| and |interfaces|.
+  void RegisterEntryPoint(const uint32_t id, SpvExecutionModel execution_model,
+                          std::vector<uint32_t>&& interfaces) {
     entry_points_.push_back(id);
-    entry_point_interfaces_.emplace(id, std::vector<uint32_t>());
     entry_point_to_execution_models_[id].insert(execution_model);
+    entry_point_interfaces_[id].emplace_back(interfaces);
   }
 
   /// Returns a list of entry point function ids
   const std::vector<uint32_t>& entry_points() const { return entry_points_; }
 
-  /// Adds a new interface id to the interfaces of the given entry point.
-  void RegisterInterfaceForEntryPoint(uint32_t entry_point,
-                                      uint32_t interface) {
-    entry_point_interfaces_[entry_point].push_back(interface);
-  }
-
   /// Registers execution mode for the given entry point.
   void RegisterExecutionModeForEntryPoint(uint32_t entry_point,
                                           SpvExecutionMode execution_mode) {
     entry_point_to_execution_modes_[entry_point].insert(execution_mode);
   }
 
-  /// Returns the interfaces of a given entry point. If the given id is not a
-  /// valid Entry Point id, std::out_of_range exception is thrown.
-  const std::vector<uint32_t>& entry_point_interfaces(
+  /// Returns the interface lists of a given entry point. If the given id is not
+  /// a valid Entry Point id, std::out_of_range exception is thrown.
```

Comment:
Hm. We build without exceptions IIRC, so throwing is not a thing.
Anyway, ok to leave that in. It was a pre-existing issue.

Reply:
Ended up changing this since I changed the method again.